### ISSUE TITLE
HELP PLEASE

### ISSUE THREAD

**Issue by mzhnq12** [Vastlab/libMR#17]
$ virtualenv venv && source venv/bin/activate
$ pip install -r python/requirements.txt
$ pip install .
Update: libmr is now available on PyPI. To install the latest version, you can simply:

$ pip install libmr
Building a shared object

If you have Cython and NumPy already installed but you do not want to install LibMR into Python's path, you can simply build the module directly:

$ python setup.py build_ext -i
This will place 'libmr.so' into the current folder. You can import it directly:

$ file libmr.so
libmr.so: ELF 64-bit LSB  shared object, ...
$ python
>>> import libmr
>>> mr = libmr.MR()
Now, move libmr.so into your project's folder, or add this path into your PYTHONPATH environment variable.

I have read your guidance when I try to install libMR，however  when I excute "python setup.py build_ext -i" there is no "libmr.so" in my folder which made me upset. I was using conda virtual environment and I did it based on windows because I do not have Linux/mac.Could you please help me to solve this problem,thank you very much.

**Comment by siebenkopf** [comment]
I am not sure why you are not using the `pip install libmr` from PyPI directly. This will automatically solve all the issues that you have with the generated files.

Note that the documentation lists two different, independent ways of building the package. These are not two dependent steps that you need to perform. A simple `pip install libmr` is completely sufficient.

Anyway, when you still want to build locally,  the `libmr.so` is typically placed into some sub-folder, for example `build`. Have you tried to find the file in these folders?

**Comment by mzhnq12** [comment]
Thank you for your reply!
I did search **libmr.so** in other folder，however，I still can not find it.
And at first I use pip to install libmr and when I use conda list to check it and  I see libmr is already there.
![osdn](https://github.com/Vastlab/libMR/assets/109964261/46f800bf-96f1-49da-8a41-fcf3342cfbf9)
But when I try to run the code on pycharm errors are given   
“Using TensorFlow backend.
LibMR not installed or libmr.so not found
Install libmr: cd libMR/; ./compile.sh”

That is why I try to build locally,I spent a lot of time on it but failed, which frustrated me a lot.
`conda create -n osdn1 python=3.6`
`activate osdn1`
`pip install libmr`

That is the code i use to install libmr,but pycharm still tell me libmr or libmr.so not found.

**Comment by siebenkopf** [comment]
Well, apparently you are on a Windows system, so the file name will not be `libmr.so` but likely something like `libmr.dll` or, as in my case, `libmr.cp310-win_amd64.pyd`. Compiling `libmr` from scratch on a Windows machine is definitely untested and should not be approached unless you know exactly what you are doing.

I have tried your `conda` approach on my machine which also has Windows operating system (I used python version 3.10). For me, the `pip install libmr` worked flawlessly, after making sure that the requirements are installed as well, which are:
`conda install numpy cython`. 

Hence, to me this rather seems to be an issue with setting up PyCharm to use the correct conda environment.

**Comment by mzhnq12** [comment]
Yes,I am on a Windows system because I do not have a linux system.I do have similar file like `libmr.cp36-win_amd64.pyd`(my python version is 3.6). And when I use `conda list `to check limbr, I found it was already in the environment,however when I run the code, errors are still there
“Using TensorFlow backend.
LibMR not installed or libmr.so not found
Install libmr: cd libMR/; ./compile.sh”

"Compiling libmr from scratch on a Windows machine is definitely untested and should not be approached unless you know exactly what you are doing."  I think this is the main reason,anyway Thank you for your patience,wish all the best!
